src/renderer/html_handlebars/hbs_renderer.rs

```diff
@@ -437,6 +437,54 @@ impl Renderer for HtmlHandlebars {
             is_index = false;
         }
 
+        // Render 404 page
+        if html_config.output_404 != Some("".to_string()) {
+            let default_404_location = src_dir.join("404.md");
+            let content_404 = if let Some(ref filename) = html_config.input_404 {
+                let path = src_dir.join(filename);
+                std::fs::read_to_string(&path).map_err(|failure| {
+                    std::io::Error::new(
+                        failure.kind(),
+                        format!("Unable to open 404 input file {:?}", &path),
+                    )
+                })?
+            } else if default_404_location.exists() {
+                std::fs::read_to_string(&default_404_location).map_err(|failure| {
+                    std::io::Error::new(
+                        failure.kind(),
+                        format!(
+                            "Unable to open default 404 input file {:?}",
+                            &default_404_location
+                        ),
+                    )
+                })?
+            } else {
+                "# 404 - Document not found\n\nUnfortunately, this URL is no longer valid, please use the navigation bar or search to continue.".to_string()
+            };
+            let html_content_404 = utils::render_markdown(&content_404, html_config.curly_quotes);
+
+            let mut data_404 = data.clone();
+            let base_url = if let Some(site_url) = &html_config.site_url {
+                site_url
+            } else {
+                warn!("HTML 'site-url' parameter not set, defaulting to '/'. Please configure this to ensure the 404 page work correctly, especially if your site is hosted in a subdirectory on the HTTP server.");
+                "/"
+            };
+            data_404.insert("base_url".to_owned(), json!(base_url));
+            data_404.insert("path".to_owned(), json!("404.md"));
```

Comment:
I'm not sure what "path" is used for, but I would guess this should be the path from the config?

Reply:
AFAICT, that path is only used to create the path to root for navigation links and script urls, i.e. it should _not_ be the config path in order for the links to work correctly.